lua/lazyvim/plugins/extras/lang/r.lua

```diff
@@ -53,15 +53,13 @@ return {
     dependencies = {
       "R-nvim/cmp-r",
     },
-    opts = function(_, opts)
-      table.insert(opts.sources.compat or {}, "cmp_r")
-
-      opts.sources.providers["cmp-r"] = vim.tbl_deep_extend(
-        "force",
-        opts.sources.providers["cmp-r"] or {},
-        { name = "cmp_r", module = "blink.compat.source" }
-      )
-    end,
+    opts = {
+      sources = {
+        compat = {
+          cmp_r = { name = "cmp_r", module = "blink.compat.source" },
+        },
```

Comment:
```suggestion
        compat = { "cmp_r" },
```
Just this should be enough.